clients/cli/src/validate.rs

```diff
@@ -0,0 +1,71 @@
+use crate::error::CliResult;
+use crate::utils::{get_account_or_exit, get_config};
+use crate::{err, err_unexpected};
+use lockbook_core::{Error as CoreError, get_path_by_id};
+use lockbook_core::service::integrity_service::{test_repo_integrity, TestRepoError, Warning};
+use uuid::Uuid;
+use lockbook_core::service::{path_service, integrity_service};
+use crate::error::ErrorKind::DocTreatedAsFolder;
+use lockbook_core::model::state::Config;
+
+pub fn validate() -> CliResult<()> {
+    get_account_or_exit();
+
+    let config = get_config();
+
+    let err = match integrity_service::test_repo_integrity(&config) {
+        Ok(warnings) => {
+            if warnings.len() == 0 {
+                return Ok(());
+            };
+
+            let mut warning_message = format!("{} warnings: ", warnings.len());
+            for w in warnings {
+                match w{
+                    Warning::EmptyFile(id) => {
+                        let path = get_path_by_id_or_err(&config, id)?;
+                        warning_message.push_str(&*format!("file at path {} is empty, ", path));
+                    }
+                    Warning::InvalidUTF8(id) => {
+                        let path = get_path_by_id_or_err(&config, id)?;
+                        warning_message.push_str(&*format!("file at path {} is contains invalid UTF8, ", path));
+                    }
+                }
+            }
+            // Remove the last two characters ', '
+            warning_message.pop();
+            warning_message.pop();
+            return Err(err!(warning_message));
```

Comment:
This is the last error I get 

```
error[E0599]: no variant or associated item named `warning_message` found for enum `error::ErrorKind` in the current scope
  --> clients/cli/src/validate.rs:38:29
   |
38 |             return Err(err!(warning_message));
   |                             ^^^^^^^^^^^^^^^ variant or associated item not found in `error::ErrorKind`
   | 
  ::: clients/cli/src/error.rs:44:9
   |
44 |         pub enum ErrorKind {
   |         ------------------ variant or associated item `warning_message` not found here

error: aborting due to previous error; 3 warnings emitted
```

I don't really understand the issue, though, since I see strings being passed to this elsewhere (I get the same issue if I remove err!)

Reply:
What you may be seeing is usages of `err_unexpected` which take a string.

Reply:
It seems err_unexpected also wants an error -- what do you think we should do here? I'm not sure it makes sense to add something new to errors.rs, since these warnings aren't exactly errors.  Can we not just return generic errors with arbitrary string messages?

Also, once I get this done, where shall I define integration tests?

Reply:
You can `eprintln` arbitrary things. Let's handle the warnings that way. I think the warnings should be newline separated, so you won't have to process the remaining `", "`.

Then you can return one error like `WarningsFound`, because we still want to return a non-zero status code.

Reply:
`core/tests` a good place to define expressive integration tests. I think the main things you want to test here is the validations, you don't have to test the CLI bits.

Reply:
Okay, I feel like I'm almost in a pretty good state -- it compiles, but now my only issue is trying to write a test.  When I run existing tests in core/tests/integrity_tests.rs I get

```
called `Result::unwrap()` on an `Err` value: UiError(CouldNotReachServer)
thread 'integrity_tests::test_name_conflict' panicked at 'called `Result::unwrap()` on an `Err` value: UiError(CouldNotReachServer)'
```